guides/source/asset_pipeline.md

```diff
@@ -21,12 +21,16 @@ What is the Asset Pipeline?
 The asset pipeline provides a framework to concatenate and minify or compress
 JavaScript and CSS assets. It also adds the ability to write these assets in
 other languages and pre-processors such as CoffeeScript, Sass and ERB.
+It allows assets in your application to be automatically combined with assets 
+from other gems. For example, jquery-rails includes a copy of jquery.js
+and enables AJAX features in Rails.
 
 The asset pipeline is technically no longer a core feature of Rails 4, it has
 been extracted out of the framework into the
 [sprockets-rails](https://github.com/rails/sprockets-rails) gem.
 
-The asset pipeline is enabled by default.
+The asset pipeline is enabled by default and is required for the normal operation 
```

Comment:
There is a discussion of this in rails/rails#23431. Maybe this is too strong a warning?  It depends on whether you see javascript and jquery as part of the normal operation of the framework.  

Thought experiment: What if we changed Rails so sprockets was off by default? If it would cause trouble in that case, then I think there is an argument to be made for a stronger warning before showing how to disable it in this case. 

But yeah, I'm wrestling with this line too, because technically it is not required for normal operation of Rails, so it seems incorrect in detail although it might be helpful in spirit.

I'll remove this line if you think it goes too far.  The notice added to Working with Javascripts is sufficient for those people using the helpers to know that they may not be able to use them without the pipeline.